docs/source/concepts/InstrumentAccessLayers.rst

```diff
@@ -117,6 +120,11 @@ ___________
 * Get the ``SpectrumInfo`` object as a const reference and use auto - ``const auto &spectrumInfo = ws->spectrumInfo();``.
 * Do not forget to add the import - ``#include "MantidAPI/SpectrumInfo.h"``.
 
+Performance Tests
+_________________
+
+Please add performance test for any algorithms that are widely used, or might be expected to have a performance increase. See `this performance test <https://github.com/mantidproject/mantid/pull/18189/files#diff-5695221d30495359738f90b83ceb0ba3>`_ added for the ``SpectrumInfo`` rollout for an example of adding such a test.
```

Comment:
I like Ian's suggestion of writing performance tests only for frequently used algorithms. That would avoid spending time unless really useful.